start-brew.sh

```diff
@@ -0,0 +1,192 @@
+#!/bin/bash
+
+# Copyright 2020, 2021 Red Hat Inc.
+
+
+# Based on the brew registry deploy process outlined in https://docs.engineering.redhat.com/display/CFC/Test
+
+
+#-----BOOTSTRAP HELPERS-----#
+source helpers.sh
+
+
+#-----PREPARATION & VARS-----#
+# Handle missing utilities, input validation, and preconfiguration
+
+#### Fix sed on MacOS - we need gsed
+OS=$(uname -s | tr '[:upper:]' '[:lower:]');
+SED="sed";
+if [ "${OS}" == "darwin" ]; then
+    SED="gsed";
+    if [ ! -x "$(command -v ${SED})"  ]; then
+       errorf "${RED}ERROR: $SED required, but not found.${CLEAR}\n";
+       printf "${BLUE}Perform \"brew install gnu-sed\" and try again.${CLEAR}";
+       exit 1;
+    fi;
+fi;
+
+#### Populate and/or Validate Parameters
+# The catalogsource's image repo and name - CatalogSource will reference 'image: ${CATALOGSOURCE_REPO}/${CATALOGSOURCE_IMAGE}:${CATALOGSOURCE_TAG}'
+# Make sure you know what you're doing if you edit the repo, this could mess up environment validation!
+CATALOGSOURCE_REPO=${CATALOGSOURCE_REPO:-"brew.registry.redhat.io/rh-osbs"};
+CATALOGSOURCE_BASE_REPO=$(echo ${CATALOGSOURCE_REPO} | awk 'BEGIN { FS = "/" } ; { print $1 }'); # Get just the base Registry for validation
+
+# The catalogsource's image name - CatalogSource will reference 'image: ${CATALOGSOURCE_REPO}/${CATALOGSOURCE_IMAGE}:${CATALOGSOURCE_TAG}'
+CATALOGSOURCE_IMAGE=${CATALOGSOURCE_IMAGE:-"iib-pub-pending"};
+
+# The catalogsoruce's image tag - CatalogSource will reference 'image: ${CATALOGSOURCE_REPO}/${CATALOGSOURCE_IMAGE}:${CATALOGSOURCE_TAG}'
+CATALOGSOURCE_TAG=${CATALOGSOURCE_TAG:-"v4.7"};
```

Comment:
How do we plan on keeping this up to date, will we just be manually entering it every time?

Reply:
Yepp - these will be manually updated periodically (the same way we maintain the default snapshot in `start.sh`) - it isn't ideal but there's precedent!

Reply:
The user is, in most cases, expected to set the CATALOGSOURCE_TAG but the CATALOGSOURCE_IMAGE shouldn't change.